### PR TITLE
[GuestConfiguration] Fix error messages, fix redundant get all assign…

### PR DESCRIPTION
**Pull request by viveklingaiah**
…ments

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MiYanni** [comment]
Awaiting https://github.com/Azure/azure-powershell/pull/8450

File: src/GuestConfiguration/GuestConfiguration/Properties/Resources.resx [hunk 1]
```diff
@@ -142,7 +142,7 @@
     <value>Invalid report Id {0}.</value>
   </data>
   <data name="InvalidRGOrVMName" xml:space="preserve">
-    <value>Invalid resource group name {0} or VM name {1}. Please check whether the resource group and VM exist in subscription.</value>
+    <value>Invalid resource group name {0} or VM name {1}. Please check whether the resource group and VM exist in subscription. Use cmdlet Select-AzSubscription to select a subscription.</value>
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Properties/Resources.resx]: Use Select-AzContext to select a context with a different subscription

**Comment by vladimir-shcherbakov** [comment]
@vivlingaiah Ping on the comments above

**Comment by vladimir-shcherbakov** [comment]
@vivlingaiah  Ping

File: src/GuestConfiguration/GuestConfiguration/Properties/Resources.resx (hunk 1 above)
Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Properties/Resources.resx]: done

**Comment by vivlingaiah** [comment]
- locally it did not generate dll-help file. Could you advise ?
untitled picture

**Comment by markcowl** [comment]
> * locally it did not generate dll-help file. Could you advise ?
>   untitled picture

so, this looks like you have a very old version of the build locally, or a bad merge. Those directories no longer exist.

File: src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1 [hunk 2]
```diff
@@ -51,7 +51,7 @@ DotNetFrameworkVersion = '4.7.2'
 # ProcessorArchitecture = ''
 
 # Modules that must be imported into the global environment prior to importing this module
-RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.2.0'; })
+RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.1.0'; })
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: 1.2.0, I believe.

File: src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1 [hunk 3]
```diff
@@ -104,14 +104,10 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Initial release of Az.GuestConfiguration module. Provides these two cmdlets.
-  * Get-AzVMGuestPolicyReport
-    * Provides compliance status of a VM in a resource group, compliance reasons.
-  * Get-AzVMGuestPolicyStatusHistory:
-    * Provides historical compliance statuses of a VM in a resource group, for a maximum of past 14 days.'
+        ReleaseNotes = 'Initial Release'
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: Comments from the changelog below should go here

File: src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatus.cs [hunk 4]
```diff
@@ -20,14 +20,15 @@ namespace Microsoft.Azure.Commands.GuestConfiguration.Cmdlets
...
 
     /// <summary>
     /// Gets Vm Guest Policy reports (GuestConfiguration policy reports)
     /// </summary>
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VMGuestPolicyReport", DefaultParameterSetName = ParameterSetNames.VmNameScope), 
-        OutputType(typeof(IList<GuestConfigurationAssignmentReport>))]
-    public class GetAzureRmVMGuestPolicyReport : GuestConfigurationCmdletBase
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzurePrefix + "VMGuestPolicyStatus", DefaultParameterSetName = ParameterSetNames.VmNameScope)]
+    [OutputType(typeof(PolicyStatusDetailed))]
+    [OutputType(typeof(PolicyStatusDetailed[]))]
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatus.cs]: do not specify the collection - powershell treats single values and collections alike, your first outputtype declaration is sufficient.

File: src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatusHistory.cs [hunk 5]
```diff
@@ -20,14 +20,14 @@ namespace Microsoft.Azure.Commands.GuestConfiguration.Cmdlets
...
 
     /// <summary>
     /// Gets Vm Guest Policy reports (GuestConfiguration policy reports)
     /// </summary>
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VMGuestPolicyStatusHistory", DefaultParameterSetName = ParameterSetNames.VmNameScope),
-        OutputType(typeof(IList<GuestConfigurationAssignmentReport>))]
-    public class GetAzureRmVMGuestPolicyStatusHistory : GuestConfigurationCmdletBase
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzurePrefix + "VMGuestPolicyStatusHistory", DefaultParameterSetName = ParameterSetNames.VmNameScope)]
+    [OutputType(typeof(PolicyStatus))]
+    [OutputType(typeof(PolicyStatus[]))]
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatusHistory.cs]: same comment

File: src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1 (hunk 2 above)
Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: changed

File: src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatus.cs (hunk 4 above)
Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatus.cs]: done

File: src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatusHistory.cs (hunk 5 above)
Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Cmdlets/GetAzVMGuestPolicyStatusHistory.cs]: done

File: src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1 (hunk 3 above)
Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: fixed

**Comment by vivlingaiah** [comment]
@markcowl  I'm still having issues with generating help file locally. Running into same error. I can test it out when we publish it to test gallery. What do you think ?

File: src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1 [hunk 6]
```diff
@@ -104,14 +104,10 @@ PrivateData = @{
...
-        ReleaseNotes = '* Initial release of Az.GuestConfiguration module. Provides these two cmdlets.
-  * Get-AzVMGuestPolicyReport
-    * Provides compliance status of a VM in a resource group, compliance reasons.
-  * Get-AzVMGuestPolicyStatusHistory:
-    * Provides historical compliance statuses of a VM in a resource group, for a maximum of past 14 days.'
+        ReleaseNotes = 'General bug fixes and Cmdlet Get-AzVMGuestPolicyReport renamed to Get-AzVMGuestPolicyStatus to keep it consistent with the Azure portal and to avoid user confusion.'
 
         # Prerelease string of this module
-        # Prerelease = ''
+        #Prerelease = 'preview'
```
Comment (markcowl) [inline src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: I don't think this should be a preview package - generally we don't want  major version packages to be preview, as the '0' major version already allows for breaking changes.

Comment (vivlingaiah) [reply src/GuestConfiguration/GuestConfiguration/Az.GuestConfiguration.psd1]: done

**Pull request merged**